### ISSUE TITLE
ValueError: number of bits must be greater than zero from time to time

### ISSUE THREAD

**Issue by arashsa** [aichaos/rivescript-python#36]
I can't always reproduce this error, but it seems to happen from time to time when using get and set in .rive files. This is the last time I got the error message:

> Traceback (most recent call last):
>   File "/Users/arashsaidi/Work/chatbot/bot/python/lib/python3.5/random.py", line 253, in choice
>     i = self._randbelow(len(seq))
>   File "/Users/arashsaidi/Work/chatbot/bot/python/lib/python3.5/random.py", line 230, in _randbelow
>     r = getrandbits(k)          # 0 <= r < 2**k
> ValueError: number of bits must be greater than zero
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "run.py", line 14, in <module>
>     reply = bot.reply("localuser", msg)
>   File "/Users/arashsaidi/Work/chatbot/bot/python/lib/python3.5/site-packages/rivescript/rivescript.py", line 999, in reply
>     return self._brain.reply(user, msg, errors_as_replies)
>   File "/Users/arashsaidi/Work/chatbot/bot/python/lib/python3.5/site-packages/rivescript/brain.py", line 82, in reply
>     reply = self._getreply(user, msg, ignore_object_errors=errors_as_replies)
>   File "/Users/arashsaidi/Work/chatbot/bot/python/lib/python3.5/site-packages/rivescript/brain.py", line 387, in _getreply
>     reply = random.choice(bucket)
>   File "/Users/arashsaidi/Work/chatbot/bot/python/lib/python3.5/random.py", line 255, in choice
>     raise IndexError('Cannot choose from an empty sequence')
> IndexError: Cannot choose from an empty sequence

**Comment by kirsle** [comment]
>  ValueError: number of bits must be greater than zero

I didn't even know that was a possible exception Python would throw.

> IndexError: Cannot choose from an empty sequence

This part could probably be avoided by making sure there's >1 reply available before calling `random.choice` (really, every RiveScript trigger should have at least one reply, except ones that just `@redirect` elsewhere, but RiveScript should still avoid calling `random.choice` on an empty list).

Are these related? Can you reproduce the error by having a trigger with no replies? (Maybe also an empty `{random}{/random}` tag could work too).

**Comment by kirsle** [comment]
## Python 3

```
Python 3.5.1 (default, Jul 10 2016, 20:36:01) 
[GCC 6.1.1 20160621 (Red Hat 6.1.1-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import random
>>> random.choice([])
Traceback (most recent call last):
  File "/usr/lib64/python3.5/random.py", line 253, in choice
    i = self._randbelow(len(seq))
  File "/usr/lib64/python3.5/random.py", line 230, in _randbelow
    r = getrandbits(k)          # 0 <= r < 2**k
ValueError: number of bits must be greater than zero

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.5/random.py", line 255, in choice
    raise IndexError('Cannot choose from an empty sequence')
IndexError: Cannot choose from an empty sequence
```

## Python 2

```
Python 2.7.12 (default, Jul 18 2016, 10:55:51) 
[GCC 6.1.1 20160621 (Red Hat 6.1.1-3)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import random
>>> random.choice([])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python2.7/random.py", line 275, in choice
    return seq[int(self.random() * len(seq))]  # raises IndexError if seq is empty
IndexError: list index out of range
```

Looks like the root cause *was* RiveScript trying to do `random.choice` on an empty list.

**Comment by kirsle** [comment]
This problem should be resolved if you update to v1.14.3.
